### ISSUE TITLE
Commands ending in backslash fail to parse

### ISSUE THREAD

**Issue by astronouth7303** [xonsh/xonsh#1704]
```
$ cd .virtualenvs\
...
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\execer.py", line 164, in _parse_ctx_free
    debug_level=(self.debug_level > 1))
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\parsers\base.py", line 324, in parse
    tree = self.parser.parse(input=s, lexer=self.lexer, debug=debug_level)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\ply\yacc.py", line 325, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\ply\yacc.py", line 1175, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\ply\yacc.py", line 187, in call_errorfunc
    r = errorfunc(token)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\parsers\base.py", line 2582, in p_error
    column=p.lexpos))
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\parsers\base.py", line 450, in _parse_error
    raise err
  File "<string>", line None
SyntaxError: C:\Users\James\AppData\Roaming\Python\Python35\site-packages\xontrib\vox_tabcomplete.xsh:2:0: ('code: ![',)
![cd .virtualenvs\]
^

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\ptk\shell.py", line 99, in push
    locs=None)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\execer.py", line 103, in compile
    transform=transform)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\execer.py", line 73, in parse
    tree, input = self._parse_ctx_free(input, mode=mode, filename=filename)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\execer.py", line 222, in _parse_ctx_free
    raise original_error
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\execer.py", line 164, in _parse_ctx_free
    debug_level=(self.debug_level > 1))
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\parsers\base.py", line 324, in parse
    tree = self.parser.parse(input=s, lexer=self.lexer, debug=debug_level)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\ply\yacc.py", line 325, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\ply\yacc.py", line 1175, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\ply\yacc.py", line 187, in call_errorfunc
    r = errorfunc(token)
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\parsers\base.py", line 2577, in p_error
    column=p.lexpos))
  File "C:\Users\James\AppData\Local\Programs\Python\Python35\lib\site-packages\xonsh\parsers\base.py", line 450, in _parse_error
    raise err
  File "<string>", line None
SyntaxError: C:\Users\James\AppData\Roaming\Python\Python35\site-packages\xontrib\vox_tabcomplete.xsh:0:0: EOF in multi-line statement
```

(Note that the backslash came from tab completion.)

**Comment by laerus** [comment]
probably related to #893 #693

**Comment by scopatz** [comment]
Potential fix in #2194

**Comment by scopatz** [comment]
This should be fixed as of #2198. Please feel free to reopen, if needed.
